build-status.el

```diff
@@ -375,8 +375,7 @@ Signals an error if the response does not contain an HTTP 200 status code."
 ;;;###autoload
 (define-minor-mode build-status-mode
   "Monitor the build status of the buffer's project."
-  :global t
```

Comment:
> I think you should cherry-pick my commit to fix the package provide 

That was fixed via #1. 😄 

Why do you need a custom build-status function

> my custom *build-status function returns nil will caused my custom mode-line error. I don't know how to fix it.

```el
(if (and (build-status--activate-mode) (null status))
```

Shouldn't that be `(not (null status))`?

> And I use custom build-status function in my custom mode-line to replace build-status-mode-line-string.

This is only to insert the the icons into the mode line? Is it possible to do this by [setting the build-status faces](https://github.com/sshaw/build-status#status-mode-line-faces)? 

If not, what I think you should do is this (untested): 
```el
(setq build-status-mode-line-string 
      '(eval 
        (propertize
         (concat
          (all-the-icons-faicon "cogs"
                                :v-adjust -0.05)
          (propertize " " 'face 'variable-pitch)
          (cond
           ((string= status "passed")
            (all-the-icons-faicon "check-circle")
            (build-status--propertize "P" status))
           ((string= status "running")
            (all-the-icons-faicon "spinner")
            (build-status--propertize "R" status))
           ((string= status "failed")
            (all-the-icons-faicon "chain-broken")
            (build-status--propertize "F" status))
           ((string= status "queued")
            (all-the-icons-faicon "ellipsis-h")
            (build-status--propertize "Q" status))
           (t
            (all-the-icons-faicon "question-circle-o")
            (build-status--propertize
             "?"
             (replace-regexp-in-string "[^a-zA-Z0-9[:space:]]+" " "
                                       (or status "unknown")))))))))
```